src/Phantoman.php

```diff
@@ -95,79 +98,73 @@ private function startServer()
 
         $this->resource = proc_open($command, $descriptorSpec, $this->pipes, null, null, array('bypass_shell' => true));
 
-        if (!is_resource($this->resource) || !proc_get_status($this->resource)['running']) {
+        if (!$this->isPhantomRunning()) {
             proc_close($this->resource);
             throw new ExtensionException($this, 'Failed to start PhantomJS server.');
         }
 
         // Wait till the server is reachable before continuing
-        $max_checks = 10;
-        $checks = 0;
-
         $this->write('Waiting for the PhantomJS server to be reachable');
-        while (true) {
-            if ($checks >= $max_checks) {
-                throw new ExtensionException($this, 'PhantomJS server never became reachable');
-                break;
-            }
-
-            if ($fp = @fsockopen('127.0.0.1', $this->config['port'], $errCode, $errStr, 10)) {
-                $this->writeln('');
-                $this->writeln('PhantomJS server now accessible');
-                fclose($fp);
-                break;
+        for ($checks = 0; $checks < self::MAX_CHECKS && !$this->isPhantomReachable(); ++$checks) {
```

Comment:
Is it necessary to call `!$this->isPhantomReachable()` both here and inside the loop?

Also, I don't think it matters as a standalone statement, but it seems like convention would be to write it as `$checks++`.